### DISCUSSION TITLE
What goes in the Output box?

### DISCUSSION THREAD

**Discussion by RangerGH** [fredemmott/streaming-remote#8] [Q&A]
Windows 10 pro.  I have installed the Elgato XL plugin on my local machine, added the streaming-remote.dll to the server (my remote OBS system).  On my local system which is running OBS I enter the title, the Server ws://xx.xx.xx.xx:9001 where xx is the IP address of the remote OBS system.  I copied the password from the remote system.

The Output box does has a drop down arrow but I can not put anything in it.  There is a thin orange line under the box if I click on the down arrow.  My local OBS is open and a profile selected.

The key on the XL shows a exclamation point in a triangle and the XL log shows this.
18:47:41.869                     StreamDeck                        ESDCustomPluginCommunication::onMessageReceived(): PI 83B302E9840665154DB3D9321144496A is now connected

And then if I press the key it's noted in the log.

I'm using the OBS NDI plugin to send from my local machine to the OBS that will do the streaming.

I've poked around in the code some but haven't found anything that helps me yet.  Any help would be appreciated.

**Comment by fredemmott** [comment]
This indicates it's not correctly connected to OBS on the remote server - the box gets automatically filled with whatever the OBS plugin tells it is is available. You might have more information available in the logs if you install the 'debug' version of the OBS plugin

**Comment by fredemmott** [comment]
It'd also be worth checking windows firewall settings on both machines

**Comment by RangerGH** [comment]
Okay.  I'll check that out.  The firewalls are good.  I checked that and even with them off it didn't work so something else is breaking it.  Thanks.

**Comment by fredemmott** [comment]
> I've poked around in the code 

If you're comfortable with this, you might find the web UI easier to debug with - the bulk of the code (typescript) is shared with the streamdeck plugin

**Comment by RangerGH** [comment]
I finally got back to this as life got in the way.  It now works using ws://myipadd:9002 with the password and I can start the streaming or recording on the remote.  This is with the Elgato Stream Deck XL plugin.  However, when I push the key to stop the streaming or recording nothing happens and recording or streaming continues.  The log shows the key presses.  I assume this should toggle the key or do I need to have two keys, one to start, one to stop?

**Comment by fredemmott** [comment]
I'm seeing some issues on OBS 26; I'll see what's going on and if I can fix them on 26.0; I'm not going to be trying with 26.1 yet due to unrelated issues (blackmagic support is broken in 26.1)

**Comment by fredemmott** [comment]
v26.1 will definitely need a rebuild, Qt version has changed

**Comment by fredemmott** [comment]
> The Output box does has a drop down arrow but I can not put anything in it. There is a thin orange line under the box if I click on the down arrow. My local OBS is open and a profile selected.

Selecting another action then going back to this one can workaround it; though in some cases it needs a rebuilt plugin. Going to start building for multiple OBS versions probably.

**Comment by fredemmott** [comment]
@RangerGH the build at https://github.com/fredemmott/streaming-remote/actions/runs/541122884 might fix things for you - the OBS .zip includes separate DLLs for multiple versions of OBS

If logged in, you probably want the 'native-obs-windows-Release' download link at the bottom of that page when signed in

**Comment by fredemmott** [comment]
> ws://xx.xx.xx.xx:9001 

I accidentally did the same thing and had the same problem :)

The streamdeck plugin needs the "WebSocket Port" (defaults to 9002), not the "TCP Port" (defaults to 9001).

**Reply by fredemmott** [reply]
@RangerGH Your issue should be fixed by:
- changing the port in streamdeck configuration to 9002
- updating to https://github.com/fredemmott/streaming-remote/releases/tag/v2.2.4

**Reply by RangerGH** [reply]
Thanks.  I'll check it out as soon as I can.
